src/praisonai/praisonai/scheduler/executor.py

```
@@ -146,11 +155,33 @@ async def tick(self) -> AsyncIterator[JobResult]:
+            atomic_claim = False
 
         for job in due_jobs:
-            result = await self._execute_one(job)
+            try:
+                result = await self._execute_one(job)
+            finally:
+                if atomic_claim:
+                    # Release the lease so it does not linger until expiry.
+                    try:
+                        self._runner.complete_run(job.id, self._owner_id)
+                    except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)